src/core/a-entity.js

```diff
@@ -23,155 +22,161 @@ var ONCE = {once: true};
+  attributeChangedCallback (attr, oldVal, newVal) {
+    var component = this.components[attr];
+
+    super.attributeChangedCallback();
+    // If the empty string is passed by the component initialization
+    // logic we ignore the component update.
+    if (component && component.justInitialized && newVal === '') {
+      delete component.justInitialized;
+      return;
     }
-  },
+    // When a component is removed after calling el.removeAttribute('material')
+    if (!component && newVal === null) { return; }
+    this.setEntityAttribute(attr, oldVal, newVal);
+  }
 
   /**
-   * Add to parent, load, play.
-   */
-  attachedCallback: {
-    value: function () {
-      var assetsEl;  // Asset management system element.
-      var sceneEl = this.sceneEl;
-      var self = this;  // Component.
+  * Add to parent, load, play.
+  */
+  connectedCallback () {
+    // Defer if DOM is not ready.
+    if (document.readyState === 'loading') {
+      document.addEventListener('DOMContentLoaded', this.connectedCallback.bind(this));
+      return;
+    }
 
-      this.addToParent();
+    AEntity.prototype.doConnectedCallback.call(this);
+  }
 
-      // Don't .load() scene on attachedCallback.
-      if (this.isScene) { return; }
+  doConnectedCallback () {
+    var assetsEl;  // Asset management system element.
+    var sceneEl = this.sceneEl;
+    var self = this;  // Component.
 
-      // Gracefully not error when outside of <a-scene> (e.g., tests).
-      if (!sceneEl) {
-        this.load();
-        return;
-      }
+    // ANode method.
+    super.connectedCallback();
 
-      // Wait for asset management system to finish before loading.
-      assetsEl = sceneEl.querySelector('a-assets');
-      if (assetsEl && !assetsEl.hasLoaded) {
-        assetsEl.addEventListener('loaded', function () { self.load(); });
-        return;
-      }
+    sceneEl = this.sceneEl;
+
+    this.addToParent();
+
+    // Don't .load() scene on attachedCallback.
+    if (this.isScene) { return; }
+
+    // Gracefully not error when outside of <a-scene> (e.g., tests).
+    if (!sceneEl) {
       this.load();
+      return;
     }
-  },
+
+    // Wait for asset management system to finish before loading.
+    assetsEl = sceneEl.querySelector('a-assets');
+    if (assetsEl && !assetsEl.hasLoaded) {
+      assetsEl.addEventListener('loaded', function () { self.load(); });
+      return;
+    }
+    this.load();
+  }
 
   /**
    * Tell parent to remove this element's object3D from its object3D.
    * Do not call on scene element because that will cause a call to document.body.remove().
    */
-  detachedCallback: {
-    value: function () {
-      var componentName;
+  disconnectedCallback () {
+    var componentName;
 
-      if (!this.parentEl) { return; }
+    if (!this.parentEl) { return; }
 
-      // Remove components.
-      for (componentName in this.components) {
-        this.removeComponent(componentName, false);
-      }
+    super.disconnectedCallback();
```

Comment:
We didn't have this call before. We call it a second time below.

Reply:
I tested the previous code, putting a console.log in `detachedCallback` and removing an a-entity element via the chrome inspector, it was called twice, I'm not sure why twice.
Here in the new code, this is called four times. Removing this line, `disconnectedCallback` is called twice, no error, tests are all passing.

Reply:
You can add a console.log exactly where you just removed the line, open the inspector, right click on the node and remove. Anyway it's the same behavior than in the previous version now.

Reply:
Ah I just understood why I have it twice, I was removing a-entity that has a child a-camera, so I'm removing two entities at the same time that's why.